### PR TITLE
WebGLGeometries: Ensure wireframes reflect the current geometry data

### PR DESCRIPTION
**Pull request by Mugen87**
Fixes the problem mentioned in https://github.com/mrdoob/three.js/issues/12771#issuecomment-434425886

`WebGLGeometries` now ensures that the wireframe attribute of a geometry is actually updated when the underlying index buffer changes. Before this fix, the wireframe attribute was created once and never updated.

I've created a fiddle with the fixed build in order to illustrate the new behavior:

https://jsfiddle.net/f2Lommf5/15691/

~The PR also does an optimization. Before the fix, `getWireframeAttribute()` always created an array object right here:~

https://github.com/mrdoob/three.js/blob/771b015e471ded9a9ce9656e6814c74c33c0904f/src/renderers/webgl/WebGLGeometries.js#L123

~Because of hoisting, it does not matter that the statement was placed after the return. The array was already created at this point. This is now fixed.~

### CODE REVIEW

**Comment by Mugen87** [comment]
Updating the position buffer is no problem since `updateWireframeAttribute()` creates a new set of indices. Even if the position buffer changes, the indices are still valid.

File: src/renderers/webgl/WebGLGeometries.js [hunk 1]
```diff
@@ -157,15 +154,42 @@ function WebGLGeometries( gl, attributes, info ) {
...
 		attributes.update( attribute, gl.ELEMENT_ARRAY_BUFFER );
 
 		wireframeAttributes[ geometry.id ] = attribute;
 
-		return attribute;
+	}
+
+	function getWireframeAttribute( geometry ) {
+
+		var geometryIndex = geometry.index;
```
Comment (mrdoob) [inline src/renderers/webgl/WebGLGeometries.js]: I would move this to after `if ( currentAttribute ) {` for readability.

Comment (Mugen87) [reply src/renderers/webgl/WebGLGeometries.js]: 👍

**Comment by Mugen87** [comment]
> Because of hoisting, it does not matter that the statement was placed after the return. The array was already created at this point. 

Oh sorry, I've made a mistake here. Because of hoisting, the variable `indices` is just declared and has the value `undefined`. The array is actually created **after** the return. So in this context, everything is fine even with the latest builds 😅

**Comment by Mugen87** [comment]
I think this PR is ready to go now 😊🚀

**Comment by Mugen87** [comment]
see https://github.com/mrdoob/three.js/issues/10638#issuecomment-509775215

**Comment by Mugen87** [comment]
Okay, I've added an additional change to ensure that the respective WebGLBuffer of an obsolete wireframe attribute is actually deleted. This change should prevent an evil memory leak^^

**Comment by mrdoob** [comment]
Late thanks! 😇

**Pull request merged**